File: 프로그래머스/2/131127. 할인 행사/할인 행사.py
def solution(want, number, discount):
    answer = 0
    want_obj = {}
    
    for i in range(len(number)):
        want_obj[want[i]] = number[i]
    
    for i in range(len(discount) - 9):
        tmp_obj = {}
        discount_array = discount[i: i + 10]
        for item in discount_array:
            if item in tmp_obj:
                tmp_obj[item] += 1
            else:
                tmp_obj[item] = 1
        
        flag = False
        for item in want:
            if item not in tmp_obj:
                flag = True
                break
            else:
                if (want_obj[item] > tmp_obj[item]):
                    flag = True
                    break
        
        if not flag:
            answer += 1
    
    
    return answer

File: 프로그래머스/2/131127. 할인 행사/할인 행사.js
function solution(want, number, discount) {
    var answer = 0;
    const ref = new Map();
    want.forEach((item, i) => ref.set(item, number[i]));
    
    for (let i = 0; i <= discount.length - 10; i++) {
        const arr = discount.slice(i, i + 10);
        
        let flag = true;
        for (const key of ref.keys()) {
            if (ref.get(key) !== arr.filter((item) => item === key).length) {
                flag = false;
                break;
            }
        }
        
        if (flag) answer++
    }
    
    return answer;
}